src/dswx_sar/dswx_comparison.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python3
+import os
+import argparse
+import numpy as np
+from osgeo import gdal
+
+COMPARE_DSWX_SAR_PRODUCTS_ERROR_TOLERANCE = 1e-6
+
+def _get_parser():
+    parser = argparse.ArgumentParser(
+        description='Compare two DSWx-SAR products',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    # Inputs
+    parser.add_argument('input_file',
+                        type=str,
+                        nargs=2,
+                        help='Input images')
+
+    return parser
+
+def _get_prefix_str(flag_same, flag_all_ok):
+    flag_all_ok[0] = flag_all_ok[0] and flag_same
+    return '[OK]   ' if flag_same else '[FAIL] '
+
+def _print_first_value_diff(image_1, image_2, prefix):
+    """
+    Print first value difference between two images.
+
+       Parameters
+       ----------
+       image_1 : numpy.ndarray
+            First input image
+       image_2: numpy.ndarray
+            Second input image
+       prefix: str
+            Prefix to the message printed to the user
+    """
+    flag_error_found = False
+    for i in range(image_1.shape[0]):
+        for j in range(image_1.shape[1]):
+            if (abs(image_1[i, j] - image_2[i, j]) <=
+                    COMPARE_DSWX_SAR_PRODUCTS_ERROR_TOLERANCE):
+                continue
+            print(prefix + f'     * input 1 has value'
+                  f' "{image_1[i, j]}" in position'
+                  f' (x: {j}, y: {i})'
+                  f' whereas input 2 has value "{image_2[i, j]}"'
+                  ' in the same position.')
+            flag_error_found = True
+            break
+        if flag_error_found:
+            break
+
+def _compare_dswx_sar_metadata(metadata_1, metadata_2):
+    """
+    Compare DSWx-SAR products' metadata
+
+       Parameters
+       ----------
+       metadata_1 : dict
+            Metadata of the first DSWx-SAR product
+       metadata_2: dict
+            Metadata of the second
+    """
+    metadata_error_message = None
+    flag_same_metadata = len(metadata_1.keys()) == len(metadata_2.keys())
+    if not flag_same_metadata:
+        metadata_error_message = (
+            f'* input 1 metadata has {len(metadata_1.keys())} entries'
+            f' whereas input 2 metadata has {len(metadata_2.keys())} entries.')
+
+        set_1_m_2 = set(metadata_1.keys()) - set(metadata_2.keys())
+        if len(set_1_m_2) > 0:
+            metadata_error_message += (' Input 1 metadata has extra entries'
+                                       ' with keys:'
+                                       f' {", ".join(set_1_m_2)}.')
+        set_2_m_1 = set(metadata_2.keys()) - set(metadata_1.keys())
+        if len(set_2_m_1) > 0:
+            metadata_error_message += (' Input 2 metadata has extra entries'
+                                       ' with keys:'
+                                       f' {", ".join(set_2_m_1)}.')
+    else:
```

Comment:
If the if clause above has executed, that means there is already a difference that was detected. In that case, I think you can early terminate this function. Consider early return.